src/test.rs

```diff
@@ -49,6 +51,30 @@ type PersonAndOpaqueSender = (Person, OpaqueIpcSender);
 type PersonAndReceiver = (Person, IpcReceiver<Person>);
 type PersonAndSharedMemory = (Person, IpcSharedMemory);
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+struct HasWeirdSerializer (Option<String>);
+
+thread_local! { static WEIRD_CHANNEL: RefCell<Option<IpcSender<HasWeirdSerializer>>> = RefCell::new(None) }
+
+impl Serialize for HasWeirdSerializer {
+    fn serialize<S>(&self, serializer: &mut S) -> Result<(), S::Error>
+        where S: Serializer
+    {
+        if self.0.is_some() {
+            WEIRD_CHANNEL.with(|chan| { chan.borrow().as_ref().unwrap().send(HasWeirdSerializer(None)).unwrap(); });
+        }
+        self.0.serialize(serializer)
+    }
+}
+
+impl Deserialize for HasWeirdSerializer {
+    fn deserialize<D>(deserializer: &mut D) -> Result<Self, D::Error>
+        where D: Deserializer
+    {
+        Ok(HasWeirdSerializer(try!(Deserialize::deserialize(deserializer))))
+    }
+}
+
```

Comment:
Personally I think it would be more readable to put all this stuff right before the (only) test case using it...